core/services/pg/connection.go

```diff
@@ -16,6 +18,26 @@ import (
 	"github.com/XSAM/otelsql"
 )
 
+var MinRequiredPGVersion = 110000
+
+func init() {
+	// from: https://www.postgresql.org/support/versioning/
+	now := time.Now()
+	if now.Year() > 2023 {
+		MinRequiredPGVersion = 120000
+	} else if now.Year() > 2024 {
+		MinRequiredPGVersion = 130000
+	} else if now.Year() > 2025 {
+		MinRequiredPGVersion = 140000
+	} else if now.Year() > 2026 {
+		MinRequiredPGVersion = 150000
+	} else if now.Year() > 2027 {
+		MinRequiredPGVersion = 160000
+	} else if now.Year() > 2028 {
+		panic("Please add new minimum required PostgreSQL versions")
+	}
```

Comment:
This seems risky. What if we have to release a patch version for something old? Instead of making a minimal change, we could be forced to also upgrade all the postgres references just to get CI to run.

Reply:
What do you recommend doing instead?

Reply:
Oh I guess the check can be skipped below, but this panic still seems problematic.

Reply:
Can we just warn when it's unavailable instead of panicking?